sdk/python/kfp/_client.py

```diff
@@ -559,7 +559,16 @@ def list_pipeline_versions(self, pipeline_id: str, page_token='', page_size=10,
     )
 
   # TODO: provide default namespace, similar to kubectl default namespaces.
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, version_id=None):
+  def run_pipeline(
+      self,
+      experiment_id: str,
```

Comment:
nit: experiment is also optional in single user mode

Reply:
Interesting, given there was no default value. If users call this method directly, they would need to pass `None`?

Reply:
Good point, we can confirm that later. Not blocking this PR